If fullscreen is not possible, open the map in a new tab

xavijam opened the pull request:
Apart from I introduced a bug last time I changed the fullscreen functionality (god), after talking with @iriberri and @oriolbx, if the fullscreen behaviour is not possible, we provide the possibility to open the map in a new tab, how does it sound? This could only happen when the map is added as an iframe.

REVIEWER: @alonsogarciapablo

xavijam commented:
retest this please

alonsogarciapablo commented:
Taking a look!

xavijam commented:
Thanks buddy!

alonsogarciapablo commented on src/geo/ui/fullscreen.js:
```diff
@@ -84,57 +68,49 @@ cdb.ui.common.FullScreen = cdb.core.View.extend({
-      $el.html(this.options.template(options));
-    } else {
-      cdb.log.info('FullScreen is deprecated on insecure origins. See https://goo.gl/rStTGz for more details.');
+    var $el = this.$el;
```
This variable is only being used once. We could use this.$el directly in line #88.

alonsogarciapablo commented on src/geo/ui/fullscreen.js:
```diff
@@ -84,57 +68,49 @@ cdb.ui.common.FullScreen = cdb.core.View.extend({
+    $el.html(this.options.template(options));
+
+    if (!this._canFullScreenBeEnabled()) {
+      this.undelegateEvents();
```
I didn't know about this... :+1:

alonsogarciapablo commented:
:+1: It's a nice workaround.

xavijam commented:
thanks! :cactus:

xavijam commented:
I'd like to cc @javisantana too.

javisantana commented on src/geo/ui/fullscreen.js:
```diff
@@ -14,47 +14,33 @@ cdb.ui.common.FullScreen = cdb.core.View.extend({
+    $(document).on('webkitfullscreenchange mozfullscreenchange fullscreenchange', function() {
+      if (!document.fullscreenElement && !document.webkitFullscreenElement && !document.mozFullScreenElement && !document.msFullscreenElement) {
+        if (self.model.get("allowWheelOnFullscreen")) {
+          mapView.options.map.set("scrollwheel", false);
```
mapView.model does not work?

xavijam replied:
I don't think so, ```mapView.map``` should work.
https://github.com/CartoDB/cartodb.js/blob/develop/src/geo/map.js#L584
In any case this is inherited code.

javisantana commented:
Another option to fix this would be to do it outside the control so if the page is not allowed to have fullscreen just not add it. 

In this case the control itself is the one dealing with the document logic.

fine with me

xavijam commented:
We were removing the component before and some people got angry because they wanted their buttons back.

javisantana commented:
¿? if it's not allowed why people get angry? you can't fix that so

xavijam commented:
As we mentioned to the users (cc @iriberri and @oriolbx) it is something we can't fix because is a browser decision. In any case, as I said at the beginning, the behaviour I implemented was not working properly in all cases.
Checking the feedback we received when that change was released and having to fix it, we talked about a new solution, taking into account 1 - fix the problem we introduced, 2 - avoiding to remove the button if FullScreen API is not allowed.
@iriberri @oriolbx and I thought in this solution, but if we don't agree, it is ok. That's why I wanted you to take part of this issue.
